### DISCUSSION TITLE
[request] Support _metadata files for parquet

### DISCUSSION THREAD

**Discussion by IanCal** [duckdb/duckdb#4462] [General]
Parquet files have a chunk of metadata in them. If you have a partitioned dataset, you can put all of this metadata in a single file, which contains links off to the 

This gets away from having to deal with specific partitioning schemes, or reading every file just to get the stats. Dask writes this out I believe, and you can create them by yourself.

https://arrow.apache.org/docs/python/parquet.html#writing-metadata-and-common-metadata-files

These are explicitly not supported by the parquet reader:

https://github.com/duckdb/duckdb/blob/2c623978d700443665d519282ee52891e4573a3d/extension/parquet/column_reader.cpp#L152

While it's possible to work around this by registering an arrow dataset in python, this can't then be used in pyodide yet (arrow compilation in wasm seems tricky) or in the WASM shell where reading from external data is more relevant.

```
D select * from parquet_scan('_metadata');
Error: Only inlined data files are supported (no references)
D select count(*) from parquet_scan('_metadata');
┌──────────────┐
│ count_star() │
├──────────────┤
│ 6            │
└──────────────┘
D load httpfs;
D select count(*) from parquet_scan('http://localhost:8000/_metadata');
┌──────────────┐
│ count_star() │
├──────────────┤
│ 6            │
└──────────────┘
D select * from parquet_scan('http://localhost:8000/_metadata');
Error: Only inlined data files are supported (no references)
D```

WASM shell fails to return an error message

```
duckdb> select * from parquet_scan('http://localhost:8000/_metadata');
┌┐
└┘
```

Sample files for quick testing:
[sample_metadata_file.zip](https://github.com/duckdb/duckdb/files/9308493/sample_metadata_file.zip)

**Comment by hannes** [comment]
Happy to review a PR!
